python/rapidsmpf/rapidsmpf/buffer/packed_data.pyx

```diff
@@ -13,3 +15,8 @@ cdef class PackedData:
         cdef PackedData self = PackedData.__new__(PackedData)
         self.c_obj = move(obj)
         return self
+
+    def __init__(self, PackedColumns packed_columns) -> None:
+        self.c_obj = make_unique[cpp_PackedData](
+            move(deref(packed_columns.c_obj).metadata),
+            move(deref(packed_columns.c_obj).gpu_data))
```

Comment:
This needs to very clearly document that it takes ownership of the data in the passed in `PackedColumns` object.